plugins/newsdownloader.koplugin/main.lua

```diff
@@ -68,13 +69,28 @@ end
 
 function NewsDownloader:addToMainMenu(menu_items)
     self:lazyInitialization()
-
     menu_items.news_downloader = {
         text = _("News (RSS/Atom) downloader"),
         sub_item_table = {
             {
                 text = _("Download news"),
-                callback = function() self:loadConfigAndProcessFeeds() end,
+                callback = function()
+                    if not NetworkMgr:isOnline() then
+                        wifi_enabled_before_action = false
+                        if nd_action_when_wifi_off() == "turn_on" then
```

Comment:
This logic has no place here. However, this does show a more valid method of prototyping new functionality that would be friendlier to mainlining: a plugin-specific setting that defaults to what it should.

Reply:
I don't understand what you mean here. 
I can move this logic deeper. But what with second sentence?

Reply:
It should be completely transparent and the logic should be part of NetworkManager. This needs to apply to the entire program.

Regarding the second sentence, I'm saying that a module/plugin randomly breaking user expectations is bad, but that giving a specific plugin an option to behave differently could be a way to enable wider testing to see if it works before mainlining it.